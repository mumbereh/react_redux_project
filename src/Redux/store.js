import { configureStore } from '@reduxjs/toolkit';
import homepageReducer from './Home/HomePageSlice';

const store = configureStore({
  reducer: {
    homepage: homepageReducer,
  },
});

export default store;
