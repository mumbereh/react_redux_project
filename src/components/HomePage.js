import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { setFilter, fetchCategories } from '../Redux/Home/HomePageSlice';
import '../style/Home.css';

const CurrencyTable = ({ categories }) => (
  <table className="currency-table">
    <colgroup>
      <col span={1} style={{ width: '5%' }} />
      <col span={1} style={{ width: '4%' }} />
    </colgroup>
    <thead>
      <tr>
        <th>Cur Logo</th>
        <th>Name</th>
        <th>Symbol</th>
        <th>Current Price</th>
        <th>Market Cap</th>
        <th>Market Cap Rank</th>
      </tr>
    </thead>
    <tbody>
      {categories.map((category) => (
        <tr key={category.id}>
          <td>
            <Link to={`/details/${category.id}`}>
              <img src={category.image.small} alt={category.name} />
            </Link>
          </td>
          <td>
            <Link to={`/details/${category.id}`}>{category.name}</Link>
          </td>
          <td>{category.symbol}</td>
          <td>{category.market_data.current_price.usd}</td>
          <td>{category.market_data.market_cap.usd}</td>
          <td>{category.market_data.market_cap_rank}</td>
        </tr>
      ))}
    </tbody>
  </table>
);

CurrencyTable.propTypes = {
  categories: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
};

const HomePage = () => {
  const filter = useSelector((state) => state.homepage.filter);
  const categories = useSelector((state) => state.homepage.categories);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchCategories());
  }, [dispatch]);

  const handleFilterChange = (event) => {
    dispatch(setFilter(event.target.value));
  };

  const filteredCategories = categories.filter(
    (category) => category.name.toLowerCase().includes(filter.toLowerCase()),
  );

  return (
    <div className="home-page">
      <h1>Foreign Currency</h1>
      <div className="search-container">
        <input
          className="search-input"
          type="text"
          placeholder="Currency name"
          value={filter}
          onChange={handleFilterChange}
        />
      </div>
      <CurrencyTable categories={filteredCategories} />
    </div>
  );
};

export default HomePage;
