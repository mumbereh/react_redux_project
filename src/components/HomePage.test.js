import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../Redux/store';
import HomePage from './HomePage';

test('Homepage matches snapshot', () => {
  const { container } = render(
    <Provider store={store}>
      <Router>
        <HomePage />
      </Router>
    </Provider>,
  );
  expect(container).toMatchSnapshot();
});
