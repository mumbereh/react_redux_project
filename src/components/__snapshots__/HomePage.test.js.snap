// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage matches snapshot 1`] = `
<div>
  <div
    class="home-page"
  >
    <h1>
      Foreign Currency
    </h1>
    <div
      class="search-container"
    >
      <input
        class="search-input"
        placeholder="Currency name"
        type="text"
        value=""
      />
    </div>
    <table
      class="currency-table"
    >
      <colgroup>
        <col
          span="1"
          style="width: 5%;"
        />
        <col
          span="1"
          style="width: 4%;"
        />
        <col
          span="1"
          style="width: 8%;"
        />
        <col
          span="1"
          style="width: 10%;"
        />
        <col
          span="1"
          style="width: 8%;"
        />
        <col
          span="1"
          style="width: 8%;"
        />
      </colgroup>
      <thead>
        <tr>
          <th>
            Cur Logo
          </th>
          <th>
            Name
          </th>
          <th>
            Symbol
          </th>
          <th>
            Current Price
          </th>
          <th>
            Market Cap
          </th>
          <th>
            Market Cap Rank
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
